source/_posts/routing-in-react.md

```diff
@@ -0,0 +1,497 @@
+To use React Router with Redux, you will need to use the `react-router-redux` library, which allows you to bind the router's state to the Redux store.
+
+To get started, you will need to install both `react-router-dom` and `react-router-redux`:
+
+```bash
+npm install react-router react-router-dom react-router-redux --save
+```
+
+Next, you will need to set up your Redux store and configure the react-router-redux middleware:
+
+```javascript
+import { createStore, combineReducers, applyMiddleware } from "redux";
+import { routerReducer, routerMiddleware } from "react-router-redux";
+import { createBrowserHistory } from "history";
+
+const history = createBrowserHistory();
+
+const middleware = routerMiddleware(history);
+
+const store = createStore(
+  combineReducers({
+    router: routerReducer,
+    // Other reducers go here
+  }),
+  applyMiddleware(middleware)
+);
+```
+
+Then, you can use the `ConnectedRouter` component from react-router-redux to bind the router to your Redux store:
+
+```javascript
+import {ConnectedRouter} from 'react-router-redux';
+
+<Provider store={store}>
+ <ConnectedRouter history={history}>
+  <App/>
+ </ConnectedRouter>
+</Provider>
+```
+
+With these changes, the router's state will now be managed by the Redux store, and you can access the router's state and dispatch navigation actions using the Redux dispatch function.
+
+## Frequently Encountered Issues
+
+There are a few common issues that you may encounter when using React Router:
+
+1. **Route matching issues**: Make sure that your routes are correctly defined and nested. The router will only render the first route that matches the current URL, so if you have multiple routes defined with the same path, only the first one will be rendered.
+
+2. **404 errors**: If you are using dynamic routes and the URL does not match any of your defined routes, the router will render a "404" page. Make sure that you have a catch-all route defined that will match any URL that does not match any of your other routes.
+
+3. **TypeScript errors**: If you are using TypeScript, you may encounter errors due to the types of the ``history`` object not being correctly inferred. To fix this, you can use the ``@types/react-router`` package to provide the correct types for React Router.
+
+4. **Server-side rendering issues**: If you are using server-side rendering with React Router, you may encounter issues
+   with the router's state not being properly hydrated on the client. To fix this, you can use the ``react-router-dom/server`` module to render the app on the server and pass the rendered HTML and the router's state to the client.
+
+## Summary
+
+React Router is a popular library for adding routing to a React application. It allows you to define routes and link to them, and it will handle rendering the correct components when the URL changes.
+To use React Router, you will need to install the ``react-router-dom`` library and wrap your application in a ``Router``component. 
```

Comment:
In summary you describe how router works using many sentences etc. You've already done that before. Keep it simple. Summary is the place where you engage readed what can do next for example. What’s the one next step readers should do if they finish this post? And as the last sentence you can for example: ask a question. It can encourage comments, and also make your reader think. Check other posts how we do it.